assets/js/lib/analytics/index.js

```diff
@@ -1,22 +1,39 @@
 import posthog from 'posthog-js';
 
 import { getFromConfig } from '@lib/config/config';
+import { getSettings } from '@lib/api/analyticsSettings';
+import { logError } from '@lib/log';
 
 const analyticsEnabled = getFromConfig('analyticsEnabled');
 const analyticsKey = getFromConfig('analyticsKey');
 const analyticsUrl = getFromConfig('analyticsUrl');
 const installationID = getFromConfig('installationID');
 
-if (analyticsEnabled) {
+export const initPosthog = () => {
   posthog.init(analyticsKey, {
     api_host: analyticsUrl,
     capture_pageview: false,
   });
-}
+};
 
 export const capture = (event, payload) => {
   if (!analyticsEnabled) {
     return;
   }
   posthog.capture(event, { ...payload, installationID });
 };
+
+// Check if Analytics is enabled
+if (analyticsEnabled) {
```

Comment:
This will be always executed when the file is imported, normally it is better to define functions to control the execution.
What do you think ?